lib/spack/spack/operating_systems/cray_backend.py

```diff
@@ -79,7 +79,10 @@ class Cnl(OperatingSystem):
     def __init__(self):
         name = 'cnl'
         version = self._detect_crayos_version()
-        super(Cnl, self).__init__(name, version)
+        if version:
+            super(LinuxDistro, self).__init__(name, version)
```

Comment:
That's nasty, here we are skipping one step in the MRO :slightly_smiling_face:

Reply:
I couldn't see a way around it to accomplish the same goal... python lets you do evil things to MRO

Reply:
this seems like a good application for an intermediate Factory to return a class with the proper mixin

Reply:
@dev-zero The intermediate factory is a meta class in Python and might be an overkill here.

Reply:
I'm ok with this but it needs a comment explaining why we do it.